api/controllers/places.js

```diff
@@ -36,7 +40,7 @@ const bindContacts = (lineage, contacts) => {
 
 const bindParents = (lineage, result) => {
   let current = result;
-  while (current) {
+  while (current && current.parent && current.parent._id) {
```

Comment:
Mostly as above, with the added note that we should have always been checking for `current.parent`